src/main/java/com/google/starfish/services/FavoriteNoteService.java

```diff
@@ -0,0 +1,152 @@
+
+/**
+ * Service class for FavoriteNotes Table
+ */
+public class FavoriteNoteService {
+
+  /** Gets a favorite note by the compoud id */
+  public ResultSet getRowByCompoundId(DataSource pool, long noteId, String userId) throws SQLException {
+    try (Connection conn = pool.getConnection()) {
+      String stmt = 
+          "SELECT * "
+        + "FROM favorite_notes "
+        + "WHERE `note_id`= ? AND "
+        + "`user_id` = ? "
+        + "LIMIT 1;";
+      try (PreparedStatement getStmt = conn.prepareStatement(stmt)) {
+        getStmt.setLong(1, noteId);
+        getStmt.setString(2, userId);
+        ResultSet rs = getStmt.executeQuery();
+        return rs;
+      } 
+    }
+  }
+
+  /** Deletes a favorite note by the compound id */
+  public boolean deleteRowByCompoundId(DataSource pool, long noteId, String userId) throws SQLException {
+    try (Connection conn = pool.getConnection()) {
+      String stmt = 
+          "DELETE * "
+        + "FROM favorite_notes "
+        + "WHERE `note_id`= ? AND "
+        + "`user_id` = ? "
+        + "LIMIT 1;";
+      try (PreparedStatement deleteStmt = conn.prepareStatement(stmt)) {
+        deleteStmt.setLong(1, noteId);
+        deleteStmt.setString(2, userId);
+        boolean deleted = deleteStmt.execute();
+        return deleted;
+      } 
+    }
+  }
+
+  /** Inserts a favorite note by noteId and userId */
+  public void insertFavoriteNote(DataSource pool, long noteId, String userId) {
+    if (userId == null) return;
+    try (Connection conn = pool.getConnection()) {
+      try {
+        conn.setAutoCommit(false);
+        String stmt =
+            "INSERT INTO favorite_notes ( "
+                + "note_id,"
+                + "user_id ) "
+          + "VALUES ( "
+                + "?,"
+                + "? ); ";
+        try (PreparedStatement insertStmt = conn.prepareStatement(stmt)) {
+          insertStmt.setLong(1, noteId);
+          insertStmt.setString(2, userId);
+          System.out.println(insertStmt);
+          insertStmt.execute();
+          conn.commit();
+        }
+      } catch(SQLException ex) {
+        if (conn != null) {
+          try {
+            System.err.print("Transaction is being rolled back.");
+            conn.rollback();
+          } catch (SQLException excep) {
+            System.err.print(excep);
+          }
+        }
+      }
+    } catch (SQLException ex) {
+      System.err.print(ex);
+    }
+  }
+
+  /** Gets an array of note objects that have been favorited by passed user id */
+  public Note[] getFavoriteNotesByUserId(DataSource pool, String userId) throws SQLException {
+    List<Note> notes = new ArrayList<>();
+    try (Connection conn = pool.getConnection()) {
+      String stmt = 
+          "SELECT  a.* "
+        + "FROM "
+          + Table.NOTES.getSqlTable() + " AS a "
+          + "INNER JOIN (SELECT * "
+                      + "FROM " + Table.FAVORITE_NOTES.getSqlTable() + " "
```

Comment:
nit: I noticed that you used `favorite_notes` in some of your earlier queries then used `Table.FAVORITE_NOTES.getSqlTable()` in this query. Since you already created the Table enum, maybe you could store `Table.FAVORITE_NOTES.getSqlTable()` in a variable and use that? Or just use `favorite_notes` in all the queries?

Same thing with MiscNoteLabelService

Reply:
Interesting, yeah I think keeping in a variable is the best approach here since we've already created the enum, I'll make those changes in both files.